src/MatrixClient.cc

```diff
@@ -1227,3 +1198,39 @@ MatrixClient::makeUploadRequest(QSharedPointer<QIODevice> iodev)
 
         return reply;
 }
+
+QJsonObject
+MatrixClient::getUploadReply(QNetworkReply *reply)
+{
+        QJsonObject object;
+
+        reply->deleteLater();
+
+        int status = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+
+        if (status == 0 || status >= 400) {
+                emit syncFailed(reply->errorString());
+                return object;
+        }
+
+        auto res_data = reply->readAll();
+
+        if (res_data.isEmpty())
+                return object;
+
+        auto json = QJsonDocument::fromJson(res_data);
+
+        if (!json.isObject()) {
+                qDebug() << "Media upload: Response is not a json object.";
+                return object;
+        }
+
+        object = json.object();
+        if (!object.contains("content_uri")) {
+                qDebug() << "Media upload: Missing content_uri key";
+                qDebug() << object;
+                return object;
```

Comment:
Returning a nonempty object that isn't actually usable will probably lead to bugs down the line.

Reply:
Good catch! This was not my code, I just moved it to its own function because it was being repeated in every file type upload. Will return an empty object instead.